### ISSUE TITLE
Please specify a key field name in the template parameters

### ISSUE THREAD

**Issue by TimAlonso** [dd4t/DD4T.TridionTemplates#33]
Populated the parameters for the DD4T's Generate Resources TBB and when it runs on Template Builder it throws the error "Please specify a key field name in the template parameters".

Had a look at where the exception is thrown and found this conditional statement in DD4T.Templates.Resources.cs:

```sh
if (!HasPackageValue(Package, "KeyFieldName"))
    throw new Exception("Please specify a key field name in the template parameters");
```

I looked for where the key field name parameter is set and found DD4T.Templates.Resources.Schemas.Resources Parameters.xsd. Line 21:

```sh
<xsd:element name="KeydFieldName" minOccurs="1" maxOccurs="1" type="xsd:normalizedString">
  <xsd:annotation>
    <xsd:appinfo>
    </xsd:appinfo>
  </xsd:annotation>
</xsd:element>
```

The key field name is spelt incorrectly.

**Comment by TimAlonso** [comment]
The parameter schema can be updated in SDL Web.

**Comment by quirijnslings** [comment]
Fixed this. Prerelease of base templates is available already (2.3.1.99-alpha)
The template installer with this fix will be released soon as 2.5.

Pull request https://github.com/dd4t/DD4T.TridionTemplates/pull/52.

**Comment by quirijnslings** [comment]
Release 2.5
